server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -191,6 +191,15 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
             currentState.getMetadata().custom(RepositoriesMetadata.TYPE)
         );
 
+        Optional<DiscoveryNode> remotePublicationDN = currentNodes.getNodes()
```

Comment:
Do we need this remotePublicationDN separately? Can we append the condition in the remoteDN logic above ?

Reply:
Yes, there can be cases where the remotePublicationDN has additional repository (eg :- routingTable) as compared to remoteDN and both the nodes are present in the cluster.